mcp/src/main/java/io/modelcontextprotocol/client/transport/HttpClientStreamableHttpTransport.java

```diff
@@ -442,8 +443,8 @@ public Mono<Void> sendMessage(McpSchema.JSONRPCMessage sentMessage) {
 					})).onErrorMap(CompletionException.class, t -> t.getCause()).onErrorComplete().subscribe();
 
 			})).flatMap(responseEvent -> {
-				if (transportSession.markInitialized(
-						responseEvent.responseInfo().headers().firstValue("mcp-session-id").orElseGet(() -> null))) {
+				String mcpSessionId = responseEvent.responseInfo().headers().firstValue("mcp-session-id").orElseGet(() -> null);
+				if (Objects.nonNull(mcpSessionId) && transportSession.markInitialized(mcpSessionId)) {
```

Comment:
>> API Note:
This method exists to be used as a java.util.function.Predicate, filter(Objects::nonNull)

I think `mcpSessionId != null` should be used here.